packages/typescript-estree/src/convert.ts

```diff
@@ -2360,7 +2360,7 @@ export class Converter {
         return this.fixExports(node, result);
       }
 
-      case SyntaxKind.FirstTypeNode: {
+      case SyntaxKind.TypePredicate: {
```

Comment:
both `SyntaxKind.FirstTypeNode` and `SyntaxKind.TypePredicate` have same value (163) now but its better to use node kind instead of "counter"